kubeflow/training-operator/gpt-neox/05-finetune-neox-mpijob.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: kubeflow.org/v2beta1
+kind: MPIJob
+metadata:
+  name: finetune-neox
+spec:
+  slotsPerWorker: 8
+  runPolicy:
+    cleanPodPolicy: Running
+  mpiReplicaSpecs:
+    Launcher:
+      replicas: 1
+      template:
+        spec:
+          # This is very much a hack. The NeoX code just stalls sometimes if
+          # all workers are not up. It doesn't exit gracefully so the launcher retries.
+          initContainers:
+            - image: busybox:1.28
+              name: sleep
+              command: [ "sleep", "30" ]
+          containers:
+            - image: navarrepratt/gpt-neox:7
```

Comment:
To create this image, I needed to made modifications to the `ElutherAI/gpt-neox` repo:
 - Dockerfile
   - Add the necessary source code
   - Add MPI ssh stuff
   - Remove the `mchorse` user stuff
 - Update links used to download the datasets
 - Add the `ElutherAI/best-download` repo as a submodule that tries to get imported by the `eval_tasks` package (even when just doing training)

What's the best way to making these changes available? Fork the repo into the CoreWeave org, then add a new `ml-images` job to build it?

Reply:
Yes, basically, fork it and then add a new `ml-images` job.

Reply:
The new ml-containers image is added here: https://github.com/coreweave/ml-containers/pull/9